src/jlapi.c

```diff
@@ -809,6 +809,33 @@ JL_DLLEXPORT uint64_t jl_cumulative_recompile_time_ns(void)
     return jl_atomic_load_relaxed(&jl_cumulative_recompile_time);
 }
 
+/**
+ * @brief Enable per-task timing.
+ */
+JL_DLLEXPORT void jl_task_timing_enable(void)
+{
+    // Increment the flag to allow reentrant callers.
+    jl_atomic_fetch_add(&jl_task_timing_enabled, 1);
+}
+
+/**
+ * @brief Disable per-task timing.
+ */
+JL_DLLEXPORT void jl_task_timing_disable(void)
+{
+    jl_atomic_fetch_add(&jl_task_timing_enabled, -1);
```

Comment:
```suggestion
    uint8_t tt_enabled = jl_atomic_load_relaxed(&jl_task_timing_enabled);
    if (tt_enabled > 0) {
        jl_atomic_fetch_add(&jl_task_timing_enabled, -1);
    }
    else {
        jl_error("task timing is already disabled");
    }
```

Reply:
Good call. Though i guess this could still go negative if two different threads try to decrement it at the same time... 🤔
If you want to be really careful you could check-and-set in a loop?
```
    uint8_t tt_enabled = jl_atomic_load_relaxed(&jl_task_timing_enabled);
    while (tt_enabled > 0) {
        jl_atomic_cmpswap(&jl_task_timing_enabled, &tt_enabled, tt_enabled-1);
    }
```
And actually, i don't think erroring sounds nice to me - i wouldn't want to get an error if i try to disable timing and it's already disabled. I think i'd just want it to silently return. 👍

Reply:
And actually, i don't think erroring sounds nice to me - i wouldn't want to get an error if i try to disable timing and it's already disabled. I think i'd just want it to silently return.